rasa/shared/core/trackers.py

```diff
@@ -813,3 +828,56 @@ def is_prev_action_listen_in_state(state: State) -> bool:
     """
     prev_action_name = state.get(PREVIOUS_ACTION, {}).get(ACTION_NAME)
     return prev_action_name == ACTION_LISTEN_NAME
+
+
+def get_trackers_for_conversation_sessions(
+    tracker: "DialogueStateTracker",
+) -> List["DialogueStateTracker"]:
+    """Generate trackers for `tracker` that are split by conversation sessions.
+
+    Args:
+        tracker: Instance of `DialogueStateTracker` to split.
+
+    Returns:
+        The trackers split by conversation sessions.
+    """
+    trackers: List["DialogueStateTracker"] = []
+    tracker_slots = tracker.slots
+
+    applied_events = tracker.applied_events(
+        include_restarts=True, include_previous_conversation_sessions=True
```

Comment:
I think these flags dilute the meaning of `applied_events` a bit. What about splitting at the restarts / session starts and calling `applied_events` on each tracker?

Reply:
great idea!